cdhweb/static_src/global/components/event-hero.scss

```diff
@@ -0,0 +1,115 @@
+/* 
+Event hero. Similar to standard hero. Key differences:
+- image ratio (TODO – maybe?)
+- image has a caption
+- text content – more of it, and it aligns vertically with the h1
+*/
+
+// has `.grid-standard`
+.event-hero {
+  --title-top-space: 40px;
+
+  margin-block-end: 56px;
+
+  // t = text content
+  // i = img + credit
+
+  // Layout if there is _no_ image:
+  grid-template-areas: 't t t t t t t t t t t t';
+
+  @include sm {
+    margin-block-end: 120px;
+  }
+
+  @include md {
+    --title-top-space: 64px;
+  }
+  @include lg {
+    --title-top-space: 80px;
+
+    grid-template-areas: '. t t t t t t t t t t .';
+  }
+  @include xl {
+    --title-top-space: 104px;
+  }
+  @include xxl {
+    --title-top-space: 160px;
+    grid-template-areas: '. t t t t t t t t t . .';
+  }
+
+  // Different layout if the hero has an image:
+  &:has(img) {
```

Comment:
I feel like :where and :has have significantly reduced the complexity of classes and css 🙌

Reply:
@phil-hodgson-1st Yeah, sometimes it definitely does, and sometimes it doesn't. 

It's real nice not to have to create `.my-component__details-p` just to set one CSS property on a paragraph.

Though then if you have two `p`s that need styling, then you have to decide if and which ones get this treatment, or else risk introducing more complex selectors to differentiate.

So I'm finding that, much like many decisions in CSS (and FED generally), it's a case-by-case decision.